base/special/log.jl

```diff
@@ -246,79 +259,82 @@ end
     ## Step 3: not required
 
     ## Step 4
-    Float32(u64 + q)
+    Float32(logb(Float32, base)*(u64 + q))
 end
 
-
-function log(x::Float64)
-    if x > 0.0
-        x == Inf && return x
-
-        # Step 2
-        if 0.9394130628134757 < x < 1.0644944589178595
-            f = x-1.0
-            return log_proc2(f)
+for (func, base) in (:log2=>Val(2), :log=>Val(:ℯ), :log10=>Val(10))
```

Comment:
Actually I think this could be done without metaprogramming, e.g. `log2(x::Float64) = _log(x, Val(2))`.